### PR TITLE
Add info about major versions to README.md and index.md

### PR DESCRIPTION
**Pull request by danielmkn**
Add condition to goreleaser
Remove condition from the pipeline to fix minor error (add version to changelog)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -17,7 +17,9 @@ on:
 jobs:
   goreleaser:
     runs-on: ubuntu-latest
-    if: startsWith(github.ref, 'refs/tags/') && github.event.base_ref == 'refs/heads/master'
+    if: |
+      startsWith(github.ref, 'refs/tags/') && github.event.base_ref == 'refs/heads/master' 
+      || startsWith(github.ref, 'refs/tags/v6*') && github.event.base_ref == 'refs/heads/v6'
```
Comment (alexhung) [inline .github/workflows/release.yml:22]: Does this support parenthesis to ensure the correct precedence is followed?

File: docs/index.md [hunk 2]
```diff
@@ -29,6 +29,11 @@ The following 3 license types (`jq .type`) do **NOT** support APIs:
 - JCR Edition
 - OSS
 
+## Versions
+
+We maintain two major versions of Terraform Provider - 6.x and 7.x. Version 6.x is compatible with the Artifactory versions 7.49.x and below,
```
Comment (alexhung) [inline docs/index.md:34]: Do we want to use `~>` sigil to highlight this?

Comment (danielmkn) [reply docs/index.md:34]: We can, why not. Added.

File: .github/workflows/release.yml (hunk 1 above)
Comment (danielmkn) [reply .github/workflows/release.yml]: Added.

**Review by alexhung — APPROVED** [review]

**Pull request merged**